src/bucket/FutureBucket.cpp

```diff
@@ -276,37 +298,47 @@ FutureBucket::startMerge(Application& app, uint32_t maxProtocolVersion,
                           << " with snap=" << hexAbbrev(snap->getHash());
 
     BucketManager& bm = app.getBucketManager();
+    auto& timer = app.getMetrics().NewTimer(
+        {"bucket", "merge-time", "level-" + std::to_string(level)});
 
     using task_t = std::packaged_task<std::shared_ptr<Bucket>()>;
-    std::shared_ptr<task_t> task =
-        std::make_shared<task_t>([curr, snap, &bm, shadows, maxProtocolVersion,
-                                  keepDeadEntries, countMergeEvents]() {
+    std::shared_ptr<task_t> task = std::make_shared<task_t>([
+        curr, snap, &bm, shadows, maxProtocolVersion, keepDeadEntries,
+        countMergeEvents, level, timeScope = timer.TimeScope()
+    ]() mutable {
```

Comment:
TIL that you can put 'mutable' there. Wonders never cease!

Reply:
(Also have initializer expressions in the capture clause? Gosh I need to keep up with new C++)